### PR TITLE
[docs] more details for adagrad/delta, clarifying usage of rho

### PR DESCRIPTION
**Pull request by maxpumperla**
@fchollet I've added a bit more info on adaptive optimizers.

If I remember correctly, `lr` and `decay` do not appear in the original Adadelta paper. One could argue that one benefit of Adadelta over Adagrad is that you _don't_ have to set `lr`. It's set to `lr=1` by default anyway.

Shall we remove those parameters altogether? I can do a follow-up PR.

### CODE REVIEW

File: keras/optimizers.py [hunk 1]
```diff
@@ -274,13 +274,16 @@ def get_config(self):
 
 
 class Adagrad(Optimizer):
-    """Adagrad optimizer.
+    """Adagrad optimizer. Adagrad is a gradient-based optimizer
```
Comment (fchollet) [inline keras/optimizers.py]: The first line (docstring summary) should be exactly one line. Introduce 2 line breaks after the first period

Comment (fchollet) [inline keras/optimizers.py]: All optimizers here are "gradient-based", not sure that's useful

File: keras/optimizers.py [hunk 2]
```diff
@@ -333,17 +336,21 @@ def get_config(self):
 
 
 class Adadelta(Optimizer):
-    """Adadelta optimizer.
+    """Adadelta optimizer. Adadelta is a more robust extension of Adagrad
```
Comment (fchollet) [inline keras/optimizers.py]: Same comment: introduce line breaks

File: keras/optimizers.py [hunk 3]
```diff
@@ -333,17 +336,21 @@ def get_config(self):
...
+    that adapts learning rates based on a moving window of gradient updates,
+    instead of accumulating all past gradients. Using Adadelta, you don't
+    even have to choose an initial learning rate.
```
Comment (fchollet) [inline keras/optimizers.py]: "don't even have to choose an initial learning rate." -> that's the dream, but whether Adadelta achieves this is debatable. Please make this statement more hypothetical

Comment (maxpumperla) [reply keras/optimizers.py]: what I meant is that `lr` doesn't appear in the original paper. so that's "achieved". of course, whether that leads to improved performance is a different story.

**Comment by maxpumperla** [comment]
@fchollet thanks for your feedback, should be good now. I've noticed a few other places in which docs can be improved. If this is appreciated at all, I could continue to work on this for a bit.

**Comment by fchollet** [comment]
You can also replicate this PR in tf.keras on the TensorFlow repository if you'd like.
